smolyay/basis.py

```diff
@@ -87,24 +127,14 @@ class ChebyshevFirstKind(BasisFunction):
 
     def __init__(self, n):
         super().__init__()
-        self._n = n
-        self._derivative_polynomial = None
-        if n > 0:
-            self._points = [-numpy.cos(numpy.pi*i/n) for i in range(n+1)]
-        else:
-            self._points = [0]
+        self.n = n
 
     @property
-    def points(self):
-        """list: Sampling points at extrema of polynomial."""
-        return self._points
+    def domain(self):
+        """list: Domain the sample points come from."""
+        return [-1, 1]
```

Comment:
If we switch to NumPy array as the return type, this would need to become like this:
```suggestion
        return np.array([-1, 1])
```